runtime/src/mips.rs

```diff
@@ -0,0 +1,721 @@
+
+        /// An emergency stop measure to kill a proposal. Governance committee can kill
+        /// a proposal at any time.
+        #[weight = SimpleDispatchInfo::FixedNormal(200_000)]
+        pub fn kill_proposal(origin, proposal_hash: T::Hash) {
+            T::CommitteeOrigin::try_origin(origin)
+                .map(|_| ())
+                .or_else(ensure_root)
+                .map_err(|_| "bad origin")?;
+
+            Self::close_proposal(proposal_hash.clone());
+        }
+
+        /// Moves a referendum instance into dispatch queue.
+        #[weight = SimpleDispatchInfo::FixedNormal(200_000)]
+        pub fn enact_referundum(origin, proposal_hash: T::Hash) {
+            T::CommitteeOrigin::try_origin(origin)
+                .map(|_| ())
+                .or_else(ensure_root)
+                .map_err(|_| "bad origin")?;
+
+            Self::prepare_to_dispatch(proposal_hash);
+        }
+
+        /// When constructing a block check if it's time for a ballot to end. If ballot ends,
+        /// proceed to ratification process.
+        fn on_initialize(n: T::BlockNumber) {
+            if let Err(e) = Self::end_block(n) {
+                sr_primitives::print(e);
+            }
+        }
+
+    }
+}
+
+impl<T: Trait> Module<T> {
+    /// Retrieve all proposals that need to be closed as of block `n`.
+    pub fn proposals_maturing_at(n: T::BlockNumber) -> Vec<(MipsIndex, T::Hash)> {
+        Self::proposal_meta()
+            .into_iter()
+            .filter(|meta| meta.end == n)
+            .map(|meta| (meta.index, meta.proposal_hash))
+            .collect()
+    }
+
+    // Private functions
+
+    /// Runs the following procedure:
+    /// 1. Find all proposals that need to end as of this block and close voting
+    /// 2. Tally votes
+    /// 3. Submit any proposals that meet the quorum threshold, to the governance committee
+    fn end_block(block_number: T::BlockNumber) -> Result {
+        sr_primitives::print("end_block");
+
+        // Find all matured proposals...
+        for (index, hash) in Self::proposals_maturing_at(block_number).into_iter() {
+            // Tally votes and create referendums
+            Self::tally_votes(index, hash.clone());
+
+            // And close proposals
+            Self::close_proposal(hash);
+        }
+
+        Ok(())
+    }
+
+    /// Summarize voting and create referendums if proposals meet or exceed quorum threshold
+    fn tally_votes(index: MipsIndex, proposal_hash: T::Hash) {
+        if let Some(voting) = <Voting<T>>::get(proposal_hash) {
+            let aye_stake = voting
+                .ayes
+                .iter()
+                .fold(<BalanceOf<T>>::zero(), |acc, ayes| acc + ayes.1);
+
+            let nay_stake = voting
+                .nays
+                .iter()
+                .fold(<BalanceOf<T>>::zero(), |acc, nays| acc + nays.1);
+
+            let net_stake = aye_stake - nay_stake;
+
+            if net_stake >= Self::quorum_threshold() {
```

Comment:
Should the logic not be:
`aye_stake > quorum_threshold && net_stake > 0`
so that we have the conditions:
1. more than the minimum quorum threshold has been staked to ayes
2. simple majority

Reply:
I guess I understood this differently but I've updated the logic. Used comparision to ensure simple majority.